server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -405,16 +403,24 @@ private static boolean handleMappingUpdateRequired(
         long version,
         UpdateHelper.Result updateResult
     ) {
-        final var mapperService = primary.mapperService();
         try {
+            final var sourceToMerge = new CompressedXContent(result.getRequiredMappingUpdate());
+            final var mapperService = primary.mapperService();
+            final long initialMappingVersion = mapperService.mappingVersion();
+            final var existingMapper = mapperService.documentMapper();
+            if (existingMapper != null && sourceToMerge.equals(existingMapper.mappingSource())) {
+                context.resetForNoopMappingUpdateRetry(initialMappingVersion);
+                return true;
+            }
```

Comment:
Do you have a scenario in mind where this will help specifically? You are comparing the dynamic mapping update with the existing mappings, when are those two going to be the same?

Reply:
Yea, after failing over to a new index when a bunch of data nodes are blasting the master all with the same update. No need to waste energy on all write threads and the master for longer than absolutely necessary IMO :)

Reply:
I don't understand how the conditional is matched in that scenario: how can the existing mappings be exactly the same as the dynamic mapping update? The existing mappings hold all the fields that have been added until then, and the new mappings hold the updates to be made following indexing of a new document. Say that we are indexing the first doc of a new index, the dynamic mapping update would contain a lot of fields, and the existing mappings would be empty?  I am sure I am missing something, can you help me better understand?

Reply:
> I don't understand how the conditional is matched in that scenario: how can the existing mappings be exactly the same as the dynamic mapping update?

I think that's when we bootstrap a new index that uses dynamic mapping updates. If you index the same kind of logs that will trigger the same mapping in parallel, you will and should get the exact same mapping here every time (like think some metrics line from Metricbeat or so).

> Say that we are indexing the first doc of a new index, the dynamic mapping update would contain a lot of fields, and the existing mappings would be empty

Jup exactly. E.g. Elasticsearch detailed node stats I'd say :P that's a couple hundred fields that would get created in the dynamic updates. No need to ship that out from write threads more often than absolutely necessary every time we roll-over an index.